R/lint.R

```diff
@@ -114,14 +100,7 @@ lint <- function(filename, linters = NULL, ..., cache = FALSE, parse_settings =
     }
```

Comment:
do this as `mapply()`?

Reply:
The list grows to `length(source_expressions$expressions) * length(linters)`. Could be a case for `outer()`...
Not related to cyclocomp though - for loops are linear.

Reply:
i don't think that's true:

```
cyclocomp::cyclocomp(function() for (i in 1:10) i)
# [1] 3
cyclocomp::cyclocomp(function() for (i in 1:10) for (j in 1:10) i*j)
# [1] 6
cyclocomp::cyclocomp(function() for (i in 1:10) for (j in 1:10) for (k in 1:10) i*j*k)
# [1] 10
cyclocomp::cyclocomp(function() for (i in 1:10) for (j in 1:10) for (k in 1:10) for (l in 1:10) i*j*k*l)
# [1] 15
```

looks fibonacci=quadratic actually

Reply:
Oh I didn't know that. Thanks.
Since we're already below 15, do you still want to try an `outer()` based approach?

Reply:
i think it'll look nicer / more "canonically" R, but not a priority